src/core/utils/misc.js

```diff
@@ -29,3 +31,26 @@ export function isValidStorageTypeValue(value) {
   value = ensureArray(value);
   return value.length && value.every(type => supportedPersistances.some(v => type === v));
 }
+
+export function forEachAsync(array, func) {
```

Comment:
Can you explain how this is different the `Promise.all()`?

Reply:
`Promise.all()` "consolidates" promises into a single promise and resolves when they have all resolved, but rejects as soon as one rejects. This just does a bunch of async operations and lets you know when they've all completed, regardless of their outcome. It's kind of like async.js's `forEach`, but with promises, instead of callbacks.

Reply:
Got it